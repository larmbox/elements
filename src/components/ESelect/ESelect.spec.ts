import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { ESelect } from '~/index';

test('mount component', () => {
  expect(ESelect).toBeTruthy();

  const component = mount(ESelect, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain('ESelect');
  expect(component.html()).toContain('EFormControl--field');
});

test('register css variables', () => {
  mount(ESelect);
  expect(
    document.getElementById('EThemeProvider--global')!.innerText
  ).toContain('ESelect');
});

test('inherit attributes', () => {
  const component = mount(ESelect, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
