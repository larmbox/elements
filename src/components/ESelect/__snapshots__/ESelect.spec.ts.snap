// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"EFormControl EFormControl--field ESelect ESelect--md ESelect--primary\\">
  <!--v-if-->
  <!--v-if-->
  <div class=\\"ESelect-inner\\">
    <div class=\\"ESelect-element\\"><select id=\\"id\\">
        <!--v-if-->
      </select><span class=\\"EIcon chevron-expand ESelect-icon\\" aria-hidden=\\"true\\"></span></div>
  </div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;
